projects/packages/jetpack-mu-wpcom/src/features/launchpad/class-launchpad-task-lists.php

```diff
@@ -152,6 +152,44 @@ public function is_task_list_enabled( $id ) {
 		return $this->load_value_from_callback( $task_list, 'is_enabled_callback', null );
 	}
 
+	/**
+	 * Check if a task list was dismissed by the user.
+	 *
+	 * @param string $id Task List id.
+	 * @return bool|null True if dismissed, false if not.
+	 */
+	public function is_task_list_dismissed( $id ) {
+		$task_list_dismissed_status = $this->get_task_list_dismissed_status();
+
+		// Return true if the task list is on the dismissed status array and its value is true.
+		return isset( $task_list_dismissed_status[ $id ] ) && $task_list_dismissed_status[ $id ];
+	}
+
+	/**
+	 * Set wether a task list is dismissed or not for a site.
+	 *
+	 * @param string $id Task List id.
+	 * @param bool   $is_dismissed True if dismissed, false if not.
+	 */
+	public function set_task_list_dismissed( $id, $is_dismissed ) {
+		$task_list_dismissed_status        = $this->get_task_list_dismissed_status();
+		$task_list_dismissed_status[ $id ] = $is_dismissed;
+
+		$launchpad_config                               = get_option( 'wpcom_launchpad_config', array() );
+		$launchpad_config['task_list_dismissed_status'] = $task_list_dismissed_status;
+		update_option( 'wpcom_launchpad_config', $launchpad_config );
+	}
+
+	/**
+	 * Get the task list visibility status for a site.
+	 *
+	 * @return array
+	 */
+	protected function get_task_list_dismissed_status() {
+		$launchpad_config = get_option( 'wpcom_launchpad_config', array() );
+		return isset( $launchpad_config['task_list_dismissed_status'] ) ? $launchpad_config['task_list_dismissed_status'] : array();
```

Comment:
Just to be extra-defensive, we may want to verify that we have an array in the option first.
```suggestion
		return isset( $launchpad_config['task_list_dismissed_status'] ) && is_array( $launchpad_config['task_list_dismissed_status'] ) ? $launchpad_config['task_list_dismissed_status'] : array();
		// Given the complexity, this may be worth splitting up into an early return check.
		if ( ! isset( $launchpad_config['task_list_dismissed_status'] ) || ! is_array( $launchpad_config['task_list_dismissed_status'] ) ) {
			return array();
		}

		return $launchpad_config['task_list_dismissed_status'];
```